node/src/accountant/big_int_db_processor.rs

```diff
@@ -890,261 +961,508 @@ mod tests {
-        let subject = BigIntDbProcessor::<DummyDao>::default();
-        let balance_change = Addition("balance", 5555);
-        let config = BigIntSqlConfig::new(
-            "insert into test_table (name, balance_high_b, balance_low_b) values (:name,:balance_high_b,:balance_low_b) on conflict (name) do",
-            Some(|table_name|format!("update {} set balance_high_b = balance_high_b + :balance_high_b, balance_low_b = balance_low_b + :whatever where name = :name",table_name)),
-                SQLParamsBuilder::default()
-                    .key("name", ":name", &"Joe")
-                    .wei_change(balance_change)
-                    .build(),
-            );
-
-        let result = subject.execute(Left(conn.as_ref()), config);
 
         assert_eq!(
             result,
-            Err(BigIntDbError(
-                //sadly, I wasn't able to get a nicer error case with a more obvious relation to the tested requirements
-                "Wei change: error after invalid upsert command for test_table of 5555 Wei to Joe with error 'NOT NULL constraint failed: test_table.balance_low_b'"
-                    .to_string()
-            ))
+            UpdateWithOverflowSummary {
+                balance_change_decomposed: (3, 9223372036854775804),
+                final_read_high_bytes: 4555 + 3 + SINGLE_UNIT_FROM_OVERFLOW,
+                final_read_low_bytes: 9223372036854775804 - 55 - SINGLE_UNIT_FROM_OVERFLOW,
+                math_operation_expected_result: BigIntDivider::reconstitute(4555, i64::MAX - 55)
+                    + i64::MAX as i128 * 4
+            }
         );
     }
 
     #[test]
-    fn insert_handles_unspecific_failures() {
-        let conn = initiate_simple_connection_and_test_table(
-            "big_int_db_processor",
-            "insert_handles_unspecific_failures",
+    fn update_with_overflow_for_subtraction_from_positive_num() {
+        let initial_high_bytes = 4555;
+        let initial_low_bytes = 55;
+        //signed just to make the test more unconditional
+        let balance_change_signed = -41;
+
+        let result = update_with_overflow_shared_test_body(
+            "update_with_overflow_for_subtraction_from_positive_num",
+            (initial_high_bytes, initial_low_bytes),
+            balance_change_signed,
+            Subtraction,
         );
-        let subject = BigIntDbProcessor::<DummyDao>::default();
-        let balance_change = Addition("balance", 4879898145125);
-        let config = BigIntSqlConfig::new(
-            "insert into test_table (name,balance_high_b,balance_low_b) values (:name,:balance_a,:balance_b) on conflict (name) do",
-            Some(|table_name |format!("update {} set balance_high_b = balance_high_b + 5, balance_low_b = balance_low_b + 10 where name = :name",table_name)),
-            SQLParamsBuilder::default()
-                .key("name", ":name", &"Joe")
-                .wei_change(balance_change)
-                .build(),
+
+        assert_eq!(
```

Comment:
This strangeness could be overlooked if you could do the 128-bit math right here in the test and show that the result of the 128-bit math was the same as what you get when you reconstitute the data from the database.

Like this:
let big_initial = 803247859038902890349457i128;
let big_addend = 89438948030294886894302i128;
let big_sum = big_initial + big_addend;
let (initial_high_bytes, initial_low_bytes) = BigIntDivider::decompose(big_initial);
...send through your process...
let actual_sum = BigIntDivider::reconstitute(resulting_high_bytes, resulting_low_bytes);
assert_eq (actual_sum, big_sum);

Something similar for subtraction would be good too.